src/main/java/spoon/support/reflect/declaration/CtPackageImpl.java

```diff
@@ -236,4 +237,12 @@ public boolean hasPackageInfo() {
 	public boolean isEmpty() {
 		return getPackages().isEmpty() && getTypes().isEmpty();
 	}
+
+	public void updateName(CtType<?> newType, String oldName) {
+		types.updateKey(oldName, newType.getSimpleName());
+	}
+
+	public void updateName(CtPackage newPackage, String oldName) {
+		packs.updateKey(oldName, newPackage.getSimpleName());
+	}
```

Comment:
These can both be package private.

Please name them more precisely as well: `updateTypeName` and `updatePackageName`. I'm of the opinion that overloads are generally not good for readability and discoverability.